apps/condo/domains/meter/schema/Meter.js

```diff
@@ -212,33 +220,50 @@ const Meter = new GQLListSchema('Meter', {
                     return addValidationError(B2C_APP_NOT_AVAILABLE)
                 }
             }
-        },
-        afterChange: async ({ context, operation, originalInput, updatedItem }) => {
+
             if (operation === 'create') {
                 const property = await Property.getOne(context, {
-                    id: updatedItem.property,
+                    id: newItem.property,
                     deletedAt: null,
                 })
-
                 if (property) {
-                    const hasMeterResourceOwnership = await MeterResourceOwner.getOne(context, {
-                        address: property.address,
-                        organization: { id: updatedItem.organization },
-                        resource: { id: updatedItem.resource },
+                    const meterResourceOwner = await MeterResourceOwner.getOne(context, {
+                        addressKey: property.addressKey,
+                        resource: { id: newItem.resource },
                         deletedAt: null,
                     })
 
-                    if (!hasMeterResourceOwnership) {
-                        await MeterResourceOwner.create(context, {
-                            dv: 1,
-                            sender: originalInput.sender,
-                            address: property.address,
-                            organization: { connect: { id: updatedItem.organization } },
-                            resource: { connect: { id: updatedItem.resource } },
-                        })
+                    if (meterResourceOwner && meterResourceOwner.organization.id !== newItem.organization) {
+                        throw new GQLError(ERRORS.METER_RESOURCE_OWNED_BY_ANOTHER_ORGANIZATION, context)
```

Comment:
You should do this check when updating `resource` or `property` too

Reply:
Great suggestion, fixed that and add test for this case